### ISSUE TITLE
invalid flag --chmod, with Copy command

### ISSUE THREAD

**Issue by enys** [hadolint/hadolint#609]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Linting file passes 
### Actual behavior
docker run --rm -i hadolint/hadolint:v2.2.0 < Dockerfile
/dev/stdin:87:37 invalid flag: --chmod

### Steps to reproduce the behavior
Create a simple dockerfile with:
```
#syntax=docker/dockerfile-upstream:master-experimental
FROM node:10.23.0 as builder
COPY --chown=node --chmod=777 Dockerfile .
```
Describe the exact steps to reproduce. If possible, provide a *minimum*
reproduction example; take into account that others do not have access
to your private images, source code, and environment.

REMOVE SENSITIVE DATA BEFORE POSTING (replace those parts with "REDACTED")
-->

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
docker run --rm -i hadolint/hadolint:v2.2.0 hadolint --version
Haskell Dockerfile Linter v2.2.0-0-ga34e5f3
```

**Dockerfile** (if relevant)

```Dockerfile
#syntax=docker/dockerfile-upstream:master-experimental
FROM node:10.23.0 as builder
COPY --chown=node --chmod=777 Dockerfile .
```

**Additional environment details (OS, stack version, etc.)**

**Comment by enys** [comment]
https://github.com/moby/buildkit/pull/1492

**Comment by lorenzo** [comment]
Is that a new thing

**Comment by enys** [comment]
It is, it was merged via https://github.com/moby/buildkit/pull/1492
I naivel tried ignoring by:
````
#syntax=docker/dockerfile-upstream:master-experimental
FROM node:10.23.0 as builder
# hadolint ignore=DL1000
COPY --chown=node --chmod=777 Dockerfile .
````
But strangely didn't work

**Comment by enys** [comment]
Thanks a lot for the work on this !

**Comment by lorenzo** [comment]
still missing some changes for hadolint and a new release

**Comment by deitch** [comment]
This issue has been recurring with various build syntaxes. I can `ignore=...` for things that have a code, but how do I disable checks for things like this that don't have a code? In this case, we added support for `COPY --chown=...`, but what about others that are not in yet?

In my current case, I am working with `ADD --keep-git-dir=true ...` and getting:

```
hadolint: invalid flag: --keep-git-dir
```

No code, so how would I disable it?

**Comment by grantbevis** [comment]
> This issue has been recurring with various build syntaxes. I can `ignore=...` for things that have a code, but how do I disable checks for things like this that don't have a code? In this case, we added support for `COPY --chown=...`, but what about others that are not in yet?
> 
> In my current case, I am working with `ADD --keep-git-dir=true ...` and getting:
> 
> ```
> hadolint: invalid flag: --keep-git-dir
> ```
> 
> No code, so how would I disable it?


I have the same issue, latest version of hadolint and vscode is showing me that `--keep-git-dir=bool` is a invalid flag however it's an official flag in the Docker documentation: https://docs.docker.com/engine/reference/builder/#adding-a-git-repository-add-git-ref-dir

**Comment by Bigouden** [comment]
> > This issue has been recurring with various build syntaxes. I can `ignore=...` for things that have a code, but how do I disable checks for things like this that don't have a code? In this case, we added support for `COPY --chown=...`, but what about others that are not in yet?
> > In my current case, I am working with `ADD --keep-git-dir=true ...` and getting:
> > ```
> > hadolint: invalid flag: --keep-git-dir
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > No code, so how would I disable it?
> 
> I have the same issue, latest version of hadolint and vscode is showing me that `--keep-git-dir=bool` is a invalid flag however it's an official flag in the Docker documentation: https://docs.docker.com/engine/reference/builder/#adding-a-git-repository-add-git-ref-dir

Same case for me
